### ISSUE TITLE
Genialis Expressions widget bugs and issues

### ISSUE THREAD

**Issue by Martin-Spendl** [biolab/orange3-bioinformatics#256]
**Describe the bug**
This issue will contain bugs encountered during testing. This will reduce the need for opening a new issue every single time.

**Desktop:**
 - OS: Windows
 - Python Version: 3.7 (currently)
 - Orange Version: 3.26
 - Add-on Version: JakaKokošar orange-bioinformatics@resolwe

**Additional context**
@JakaKokosar is the target audience for this issues.

**Comment by Martin-Spendl** [comment]
**Link to expressions platform in browser**

When pressing a link to the platform in a browser it uses an app.genialis/**expressions**/... no matter what server did you log into. That's why you can't see your samples in the collection.

**Comment by Martin-Spendl** [comment]
**Preview of all collections in the table of the widget**

No matter which server I log in to, it always shows every single collection there is. 

Would it be possible to only see the ones that are on the server that a user logs in to? (This kind of links to the previous issue as well)

**Comment by JakaKokosar** [comment]
> **Preview of all collections in the table of the widget**
> 
> No matter which server I log in to, it always shows every single collection there is.
> 
> Would it be possible to only see the ones that are on the server that a user logs in to? (This kind of links to the previous issue as well)

@Martin-Spendl Hmm, I don't understand what you mean. Users can choose between for example app.genialis.com or imaps.genialis.com. All the collections that are available to that user are shown in the widget. Do you get the same results if you log into either of the servers?

**Comment by Martin-Spendl** [comment]
I might have used the term 'SERVER' wrong, I meant community. I'll try to log into the other servers to test there too.

**Comment by JakaKokosar** [comment]
@mstajdohar This wasn't part of the spec or have I missed something? Should this be supported?

**Comment by Martin-Spendl** [comment]
**Z-score normalization**

Don't know if it was intended this way. Zscore normalizes expressions for each gene, but the widget normalizes across a whole sample. Basically it's working across rows instead of columns.

Log normalization works fine.

![slika](https://user-images.githubusercontent.com/61875530/87914605-29d28880-ca71-11ea-8cd3-46fb1dfd5987.png)

**Comment by mstajdohar** [comment]
@Martin-Spendl good catch!

**Comment by Martin-Spendl** [comment]
**Links to collections on the platform in a browser - AFTER UPDATE**

Looks a lot better and more convenient to work with now. Links to collections are almost right:
Instead of _app.genialis.com/**expressions**/thinc/_, it should just drop the **expression** part, so it looks like this:  _app.genialis.com/thinc/_ 

Otherwise looks nice. I'll still be searching for minor bugs and inconveniences.

**Comment by Martin-Spendl** [comment]
**Additional search filters are squashed on Windows**

Pressing _Narrow your search_ shows a few options to filter collections, but the input fields are smaller than the letter themselves.
Selecting options for sorting collections does not show the selected option (don't know if it was even intended). It sorts collections, though.

![slika](https://user-images.githubusercontent.com/61875530/88030404-78e4f000-cb3b-11ea-9c47-2deec4f44ece.png)

**Comment by JakaKokosar** [comment]
@Martin-Spendl  This happened after the latest commits?

**Comment by Martin-Spendl** [comment]
> 
> 
> @Martin-Spendl This happened after the latest commits?

Actually, I think not. I haven't been really looking into search filter, to be honest.

**Comment by JakaKokosar** [comment]
> > @Martin-Spendl This happened after the latest commits?
> 
> Actually, I think not. I haven't been really looking into search filter prior to this commit.

Interesting... on my windows machine field layout is OK. Can you check which version of PyQt you are using when running orange? And maybe try to upgrade it?

**Comment by JakaKokosar** [comment]
So the problem seems to be with windows scaling. This screenshot was taken on a 4K monitor with 150% scaling. But every widget is affected by this so I don't how to proceed with this at this moment.

<img width="1122" alt="123" src="https://user-images.githubusercontent.com/15876321/88085810-a6a35680-cb86-11ea-9d1c-50c9f6fb308d.PNG">

**Comment by mstajdohar** [comment]
I gathered my thoughts on gene expression normalization. Please support the following normalization methods (and use given labels):
- Quantile normalization
- Quantile transform to normal (row-wise)
- Quantile transform to normal (column-wise)
- Quantile transform to uniform (row-wise)
- Quantile transform to uniform (column-wise)
- Z-score (row-wise) 
- Z-score (column-wise)
- Log2(x + 1)

Quantile normalization should be implemented as defined on the [wiki page](https://en.wikipedia.org/wiki/Quantile_normalization). I did not find an implementation in Python yet, but the approach is commonly used for the normalization of gene expression data in R.

For quantile transforms use [quantile_transform](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.quantile_transform.html) method from sklearn. Set argument `n_quantiles=min(len(X), 100)`. The methods are listed in order as they should be applied if multiple are selected.

Log2 should be selected by default.

**Comment by Martin-Spendl** [comment]
**More than one expressions data object for a single sample**

When a sample has featureCounts data object as well as RSEM, it trys to get both expression data and it outputs an error, because RSEM ouputs expressions based on their 'Transcript' and not 'Gene'.
@mstajdohar what do you think about those collections (samples).

Expressions:
![slika](https://user-images.githubusercontent.com/61875530/89756998-54e65000-dae4-11ea-89e3-c1ff759ced4d.png)

Error message:
![slika](https://user-images.githubusercontent.com/61875530/89756807-d4bfea80-dae3-11ea-949e-a7c7368d11f2.png)

**Comment by mstajdohar** [comment]
This is a very important point. I will discuss with Jaka how to address it.
